prow/cmd/plank/main.go

```diff
@@ -100,37 +98,47 @@ func main() {
 		logrus.WithError(err).Fatal("Error starting config agent.")
 	}
 
-	var oauthSecret string
 	var err error
-	if o.githubTokenFile != "" {
-		oauthSecretRaw, err := ioutil.ReadFile(o.githubTokenFile)
+	// Check if github endpoint is a valid url.
+	for _, ep := range o.githubEndpoint.Strings() {
+		_, err = url.ParseRequestURI(ep)
 		if err != nil {
-			logrus.WithError(err).Fatalf("Could not read oauth secret file.")
+			logrus.WithError(err).Fatalf("Invalid --endpoint URL %q.", ep)
 		}
+	}
 
-		for _, ep := range o.githubEndpoint.Strings() {
-			_, err = url.Parse(ep)
-			if err != nil {
-				logrus.WithError(err).Fatalf("Invalid --endpoint URL %q.", ep)
+	var ghc plank.GitHubClient
+	if o.githubTokenFile != "" {
+		secretAgent := &config.SecretAgent{}
+		if err := secretAgent.Start([]string{o.githubTokenFile}); err != nil {
+			logrus.WithError(err).Fatal("Error starting secrets agent.")
+		}
+
+		getSecret := func(secretPath string) func() []byte {
```

Comment:
Almost feels like we should expose a secret getter from the agent since you've written this a lot